homeassistant/components/doods/image_processing.py

```diff
@@ -0,0 +1,366 @@
+                    dconfig[label_name] = label_confidence
+
+                # Label area
+                label_area = label.get(CONF_AREA)
+                self._label_areas[label_name] = [0, 0, 1, 1]
+                if label_area:
+                    self._label_areas[label_name] = [
+                        label_area.get(CONF_TOP),
+                        label_area.get(CONF_LEFT),
+                        label_area.get(CONF_BOTTOM),
+                        label_area.get(CONF_RIGHT),
+                    ]
+            else:
+                if label not in detector["labels"] and label != "*":
+                    _LOGGER.warning("Detector does not support label %s", label)
+                    continue
+                self._label_areas[label] = [0, 0, 1, 1]
+                if label not in dconfig or dconfig[label] > confidence:
+                    dconfig[label] = confidence
+
+        if not dconfig:
+            dconfig["*"] = confidence
+
+        # Handle global detection area
+        self._area = [0, 0, 1, 1]
+        area_config = config.get(CONF_AREA)
+        if area_config:
+            self._area = [
+                area_config.get(CONF_TOP),
+                area_config.get(CONF_LEFT),
+                area_config.get(CONF_BOTTOM),
+                area_config.get(CONF_RIGHT),
+            ]
+
+        template.attach(hass, self._file_out)
+
+        self._dconfig = dconfig
+        self._matches = {}
+        self._total_matches = 0
+        self._last_image = None
+
+    @property
+    def camera_entity(self):
+        """Return camera entity id from process pictures."""
+        return self._camera_entity
+
+    @property
+    def name(self):
+        """Return the name of the image processor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the entity."""
+        return self._total_matches
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+            ATTR_MATCHES: self._matches,
+            ATTR_SUMMARY: {
+                label: len(values) for label, values in self._matches.items()
+            },
+            ATTR_TOTAL_MATCHES: self._total_matches,
+        }
+
+    def _save_image(self, image, matches, paths):
+        img = Image.open(io.BytesIO(bytearray(image))).convert("RGB")
+        img_width, img_height = img.size
+        draw = ImageDraw.Draw(img)
+
+        # Draw custom global region/area
+        if self._area != [0, 0, 1, 1]:
+            draw_box(
+                draw, self._area, img_width, img_height, "Detection Area", (0, 255, 255)
+            )
+
+        for label, values in matches.items():
+
+            # Draw custom label regions/areas
+            if label in self._label_areas and self._label_areas[label] != [0, 0, 1, 1]:
+                box_label = "{} Detection Area".format(label.capitalize())
```

Comment:
Please use f-string.
```suggestion
                box_label = "{label.capitalize()} Detection Area"
```